pkg/plugin/ebpfwindows/ebpf_windows.go

```diff
@@ -0,0 +1,279 @@
+}
+
+// metricsMapIterateCallback is the callback function that is called for each key-value pair in the metrics map.
+func (p *Plugin) metricsMapIterateCallback(key *MetricsKey, value *MetricsValues) {
+	if key == nil {
+		p.l.Error("MetricsMapIterateCallback key is nil")
+		return
+	}
+	if value == nil {
+		p.l.Error("MetricsMapIterateCallback value is nil")
+		return
+	}
+	if key.IsDrop() {
+		p.l.Debug("MetricsMapIterateCallback Drop", zap.String("key", key.String()))
+		if key.IsEgress() {
+			metrics.DropBytesGauge.WithLabelValues(key.DropForwardReason(), egressLabel).Set(float64(value.BytesSum()))
+			metrics.DropPacketsGauge.WithLabelValues(key.DropForwardReason(), egressLabel).Set(float64(value.Sum()))
+		} else if key.IsIngress() {
+			metrics.DropBytesGauge.WithLabelValues(key.DropForwardReason(), ingressLabel).Set(float64(value.BytesSum()))
+			metrics.DropPacketsGauge.WithLabelValues(key.DropForwardReason(), ingressLabel).Set(float64(value.Sum()))
+		}
+	} else {
+		p.l.Debug("MetricsMapIterateCallback Forward", zap.String("key", key.String()))
+		if key.IsEgress() {
+			metrics.ForwardPacketsGauge.WithLabelValues(egressLabel).Set(float64(value.Sum()))
+			metrics.ForwardBytesGauge.WithLabelValues(egressLabel).Set(float64(value.BytesSum()))
+		} else if key.IsIngress() {
+			metrics.ForwardPacketsGauge.WithLabelValues(ingressLabel).Set(float64(value.Sum()))
+			metrics.ForwardBytesGauge.WithLabelValues(ingressLabel).Set(float64(value.BytesSum()))
+		}
+	}
+}
+
+// eventsMapCallback is the callback function that is called for each value  in the events map.
+func (p *Plugin) eventsMapCallback(data unsafe.Pointer, size uint32) {
+	err := p.handleTraceEvent(data, size)
+	if err != nil {
+		p.l.Error("Error handling trace event", zap.Error(err))
+	}
+}
+
+func (p *Plugin) addEbpfToPath() error {
+	currPath := os.Getenv("PATH")
+	if strings.Contains(currPath, "ebpf-for-windows") {
+		return nil
+	}
+	programFiles := os.Getenv("ProgramFiles")
+	ebpfWindowsPath := programFiles + "\\ebpf-for-windows\\"
+	newPath := currPath + ";" + ebpfWindowsPath
+	if err := os.Setenv("PATH", newPath); err != nil {
+		p.l.Error("Error setting PATH environment variable", zap.Error(err))
+		return fmt.Errorf("error setting PATH environment variable: %v", err)
+	}
+
+	return nil
+}
+
+func (p *Plugin) pullMetricsAndEvents(ctx context.Context) {
+	eventsMap := NewEventsMap()
+	metricsMap := NewMetricsMap()
+
+	err := p.addEbpfToPath()
+	if err != nil {
+		return
+	}
+
+	if enricher.IsInitialized() && p.cfg.EnablePodLevel {
+		p.enricher = enricher.Instance()
+	} else {
+		p.l.Warn("retina enricher is not initialized")
+	}
+
+	if p.enricher != nil {
+		err := eventsMap.RegisterForCallback(p.l, p.eventsMapCallback)
+		if err != nil {
+			p.l.Error("Error registering for events map callback", zap.Error(err))
+			return
+		}
+
+		defer func() {
+			p.l.Error("ebpfwindows plugin canceling", zap.Error(ctx.Err()))
+			eventsMap.UnregisterForCallback()
+		}()
+	}
```

Comment:
I believe this whole thing is required only when EnablePodLevel  is true, It won't make sense to log the warning msg when `EnablePodLevel` is false

```suggestion
	if p.cfg.EnablePodLevel {
		if enricher.IsInitialized() {
			p.enricher = enricher.Instance()
		} else {
			p.l.Warn("retina enricher is not initialized")
		}

		if p.enricher != nil {
			err := eventsMap.RegisterForCallback(p.l, p.eventsMapCallback)
			if err != nil {
				p.l.Error("Error registering for events map callback", zap.Error(err))
				return
			}

			defer func() {
				p.l.Error("ebpfwindows plugin canceling", zap.Error(ctx.Err()))
				eventsMap.UnregisterForCallback()
			}()
		}
	}
```